File: Algorithms/src/ywy/java/level1/DivisorCountPlus.java
package ywy.java.level1;

public class DivisorCountPlus {

	public DivisorCountPlus() {
	}

	public int solution(int left, int right) {
		int[] divisers = new  int[right -left + 1];
		boolean[] plusFlag = new boolean[right -left + 1];
		int index = 0;
		while(left <= right) {
			divisers[index] = left;
			plusFlag[index++] = diviserFind(left); 
		    left++;
		}
		return result(divisers, plusFlag);
	}

	private boolean diviserFind(int num) {
        if (num == 1) {
			return false;
		} else if (num <= 3) {
			return true;
		}
		int left = 2;
		int divinum = (num / 2) + 1;
		int right = divinum;
		int count = 2;
		while(true) {
			if (left == divinum) {
				if (count % 2 == 0) {
					return true;
				}
				return false;
			}
			if (left * right == num) {
				count++;
                if (left != right) {
                    count++;
                }
			}
			if (right == left) {
				right = divinum;
				left++;
				continue;
			}
			right--;
		}
	}

	private int result(final int[] divisers, final boolean[] plusFlag) {
		int index = 0;
		int range = divisers.length;
		int answer = 0;
		int temp = 0;
		do {
			temp = divisers[index];
			answer += plusFlag[index] ? temp: (~temp) + 1;
		} while(++index < range);
		return answer;
	}
	
}

File: Algorithms/src/ywy/javascript/level1/DivisorCountPlus.js
const solution = (left, right) => {
    let answer = 0;
    debugger;
    while(true) {
        if (left > right) {
            return answer;
        }
        answer += plusFlag(left) ? left: (~left) + 1;
        left++;
    }
};

const plusFlag = (num) =>{
    if ( num === 1 ) {
        return false;
    } else if ( num < 4 ) {
        return true;
    }
    let count = 2;
    let left = 2;
    let temp = Math.ceil(num / 2);
    let right = temp;
    while(true) {
        if (left === temp) {
            return count % 2 === 0;
        }
        if (left * right === num) {
            count++;
            if (left != right) {
                count++;
            }
        }
        if (left === right) {
            right = temp;
            left++;
        }
        right--;
    }
};
solution(13, 17);